lib/statement.js

```diff
@@ -0,0 +1,469 @@
+const { dbstmt, SQL_SUCCESS, SQL_NO_DATA_FOUND } = require('idb-connector');
+const Connection = require('./connection');
+
+/**
+ * @class Statement
+ * @constructor
+ * @param {Object} [connection] - A Connection Object to initialize the Statement
+ * If a connection is not provided one will be initialized for the statement
+ */
+class Statement {
+  constructor(connection = new Connection().connect()) {
```

Comment:
I get that this is how it is done currently and we should maintain that for now, but it does seem _a little_ backwards that you can create a connection _ad hoc_ by creating a new Statement. Seems like this is one case where its better to be a little more verbose and explicitly create a Connection. (which would also solve your circular dependency with Connection)

Reply:
Here it setting default value for connection to be a new Connection. 

This will only occur when a Statement constructor is called without parameters.

A new connection is created implicitly which tries to connect to '*LOCAL' as the current user.

If it cannot connect it will fail out.

As far as the circular dependency I worked around that by moving require() statement for Statement withing `connection.getStatement()`.

If we decided to remove Statement() creating an implicitly connection to only accept connection object that would resolve the circular dependency.